fs-api-blockio/src/System/FS/BlockIO/API.hs

```diff
@@ -44,7 +44,7 @@ data HasBlockIO m h = HasBlockIO {
     --
     -- Using 'submitIO' after 'close' should thrown an 'FsError' exception. See
     -- 'mkClosedError'.
-    close    :: HasCallStack => m ()
+    close    :: m ()
```

Comment:
Out of interest, why did you remove the constraint?

Reply:
Because of `bracket`:

```
• Couldn't match type: HasCallStack => IO ()
                 with: IO c0
  Expected: HasBlockIO IO HandleIO -> IO c0
    Actual: HasBlockIO IO HandleIO -> HasCallStack => IO ()
• In the second argument of ‘bracket’, namely ‘close’
  In the expression: bracket (ioHasBlockIO hfs params) close action
  In an equation for ‘withIOHasBlockIO’:
      withIOHasBlockIO hfs params action
        = bracket (ioHasBlockIO hfs params) close actiontypecheck(-Wdeferred-type-errors)
```

But, we can keep the `HasCallStack` constraint because I've found a different implementation that works:

```diff
-withIOHasBlockIO hfs params action = bracket (ioHasBlockIO hfs params) close action
+withIOHasBlockIO hfs params action =
+    bracket (ioHasBlockIO hfs params) (\HasBlockIO{close} -> close) action
```

Reply:
`(\hbio -> close hbio)` also works

Reply:
Ah, interesting.